### PR TITLE
[PPWM-12] Use Storage Transfer Service to Delete Source Objects After Transfer

### PR DESCRIPTION
**Pull request by ehigham**
RR: https://broadworkbench.atlassian.net/browse/PPWM-12
Expose object deletion options via `createTransferJob` .
Added test point though these have to be run manually.

**PR checklist**
- [ ] For each library you've modified here, decide whether it requires a major, minor, or no version bump. (Click [here](/broadinstitute/workbench-libs/blob/develop/CONTRIBUTING.md) for guidance)

If you're doing a **major** or **minor** version bump to any libraries:
- [ ] Bump the version in `project/Settings.scala` `createVersion()`
- [ ] Update `CHANGELOG.md` for those libraries
- [ ] I promise I used `@deprecated` instead of deleting code where possible

In all cases:
- [ ] Replace the appropriate version hashes in `README.md` and the `CHANGELOG.md` for any libs you changed with `TRAVIS-REPLACE-ME` to auto-update the version string
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green (CI _and_ coverage tests)
- [ ] Squash commits and **merge to develop**
- [ ] Delete branch after merge

### CODE REVIEW

**Review by dvoet — APPROVED** [review]

File: google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleStorageTransferService.scala [hunk 1]
```diff
@@ -51,6 +54,35 @@ object GoogleStorageTransferService {
       else Left(s"""Illegal job name - "$name" must start with "$prefix".""")
   }
 
+  final case class JobTransferOptions(whenToOverwrite: ObjectOverwriteOption = OverwriteObjectsIfDifferent,
+                                      whenToDelete: ObjectDeletionOption = NeverDeleteSourceObjects
+  )
+
+  sealed trait ObjectOverwriteOption
```
Comment (Qi77Qi) [inline google2/src/main/scala/org/broadinstitute/dsde/workbench/google2/GoogleStorageTransferService.scala]: we typically add `extends Product with Serializable` to `sealed trait`

File: google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleStorageTransferServiceSpec.scala [hunk 2]
```diff
@@ -90,36 +91,40 @@ class GoogleStorageTransferServiceSpec extends AsyncFlatSpec with Matchers with
     }
   }
 
+  implicit class PollTransferOperations(sts: GoogleStorageTransferService[IO]) {
+    def await(jobName: JobName): IO[Unit] = IO.sleep(5 seconds).untilM_ {
+      sts.listTransferOperations(jobName, googleProject).map {
```
Comment (Qi77Qi) [inline google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleStorageTransferServiceSpec.scala:96]: I don't know much about tranfer service sdk, but in compute sdk, GCP is returning a `com.google.api.gax.longrunning.OperationFuture`, and this class has a `get` function that'll just block until the job is done

**Review by Qi77Qi — APPROVED** [review]

File: google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleStorageTransferServiceSpec.scala (hunk 2 above)
Comment (ehigham) [reply google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleStorageTransferServiceSpec.scala:96]: Interesting. I'll take a look!

Comment (ehigham) [reply google2/src/test/scala/org/broadinstitute/dsde/workbench/google2/GoogleStorageTransferServiceSpec.scala:96]: In STS we get back `com.google.longrunning.Operation` which are just protobuf objects that don't seem to do anything clever sadly. Thanks for raising this, though,

**Pull request merged**